core/src/smartcontracts/isi/query.rs

```diff
@@ -382,7 +384,7 @@ mod tests {
             let asset_definition_id = AssetDefinitionId::from_str("rose#wonderland")?;
             assert!(domain
                 .add_asset_definition(
-                    AssetDefinition::new(asset_definition_id, AssetValueType::Quantity, true),
+                    AssetDefinition::new(asset_definition_id).quantity().build(),
```

Comment:
looking at this, I see that `AssetDefinition::new(id)` is **always** followed by `store()`/`quantity()`/etc. and my instinct is to just get rid of `new(id)` and have it like `AssetDefinition::store(id)`/ `AssetDefinition::quantity(id)`/etc. These constructors can return `WithMandatoryFieldsStruct`, I'd just remove `WithId` struct

Reply:
And to be in line with other data model structures I would rename `WithMandatoryFieldsStruct` to `NewAssetDefinition`

Reply:
> and my instinct is to just get rid of new(id) and have it like AssetDefinition::store(id)/ 

Funny how this is exactly what we started with. At least we don't have an extra issue clogging our backlog.